docs/docs/slot-validation-actions.mdx

```diff
@@ -0,0 +1,101 @@
+---
+id: slot-validation-actions
+sidebar_label: Slot Validation Actions
+title: Slot Validation Actions
+abstract: A slot validation action is a special type of custom action, designed to handle custom extraction and/or validation of slot values. This can be used to validate slots with predefined mappings or extract slots with custom mappings.
+---
+
+Rasa SDK provides three base classes that you can use to inherit from in order to define your custom slot extraction and
+validation code:
+
+## `Action` class
+
+You can use this class to build a [custom action](./custom-actions.mdx) that is limited to returning only
+`SlotSet` and `BotUttered` events. Any other event type will be filtered out by the default action
+[`action_extract_slots`](./default-actions.mdx#action_extract_slots).
+The name of the custom action will have to be specified in the `action` key of the
+[custom slot mapping](./domain.mdx#custom-slot-mappings) in the domain.
+Note that the action name must be listed in the domain `actions` section too.
+
+## `ValidationAction` class
+
+You can extend the `ValidationAction` in the Rasa SDK to define custom extraction and / or validation of slots that
+can be set or updated outside of a form context.
+With this option, you do not need to specify the `action` key in the [custom slot mapping](./domain.mdx#custom-slot-mappings).
+
+Note that you do not need to implement the `name` method in the custom action extending `ValidationAction`, since this
+is already provided: `action_validate_slot_mappings`. However this name must be added to the domain `actions` list.
```

Comment:
What happens if a user overrides the name? It never gets called right? If so worth mentioning here

Reply:
Yes, that's correct, the custom validation action will never run, I'll add a comment.